File: translations/codenet/deepseek_commented/C++2J/granite/Code/s571001771.java

import java.util.*;
import java.io.*;

class s571001771 {
  public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    String[] nm = br.readLine().split(" ");
    int n = Integer.parseInt(nm[0]);
    int m = Integer.parseInt(nm[1]);
    long[] key = new long[m+1];
    long[] cost = new long[m+1];
    long[][] dp = new long[(1<<12)+5][m+1];
    for(int i=0;i<(1<<12)+5;i++){
      for(int j=0;j<m+1;j++){
        dp[i][j] = Long.MAX_VALUE;
      }
    }
    dp[0][0] = 0;
    for(int i=1;i<=m;i++){
      String[] bc = br.readLine().split(" ");
      cost[i] = Long.parseLong(bc[0]);
      int b = Integer.parseInt(bc[1]);
      for(int j=0;j<b;j++){
        int c = Integer.parseInt(br.readLine());
        key[i] += (1<<(c-1));
      }
    }
    for(int i=0;i<(1<<n);i++){
      for(int j=0;j<m;j++){
        dp[i][j+1] = Math.min(dp[i][j],dp[i][j+1]);
        dp[i|key[j+1]][j+1] = Math.min(dp[i|key[j+1]][j+1],dp[i][j]+cost[j+1]);
      }
    }
    if(dp[(1<<n)-1][m]==Long.MAX_VALUE)System.out.println(-1);
    else System.out.println(dp[(1<<n)-1][m]);
  }
}

Translate the above Java code to Python and end with comment "

File: translations/codenet/uncommented/C++2P/gpt-4o/Code/s571001771.py
import sys

INF = 123456789012

def main():
    n, m = map(int, input().split())
    dp = [[INF] * (m + 1) for _ in range(1 << n)]
    dp[0][0] = 0

    cost = [0] * (m + 1)
    key = [0] * (m + 1)

    for i in range(1, m + 1):
        b = int(input())
        cost[i], b = map(int, input().split())
        for _ in range(b):
            c = int(input())
            key[i] |= (1 << (c - 1))

    for i in range(1 << n):
        for j in range(m):
            dp[i][j + 1] = min(dp[i][j], dp[i][j + 1])
            dp[i | key[j + 1]][j + 1] = min(dp[i | key[j + 1]][j + 1], dp[i][j] + cost[j + 1])

    if dp[(1 << n) - 1][m] == INF:
        print(-1)
    else:
        print(dp[(1 << n) - 1][m])

if __name__ == "__main__":
    main()

# <END-OF-CODE>
